src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java

```diff
@@ -101,7 +106,7 @@ public SenderRecipientType createSenderRecipient(KorrespondansepartResource inpu
                 .map(Link::getHref)
                 .map(s -> StringUtils.substringAfterLast(s, "/"))
                 .map(StringUtils::upperCase)
-                .filter(s -> StringUtils.equalsAny(s, "EA", "EM", "EK", "IA", "IM", "IK"))
+                .filter(s -> StringUtils.equalsAny(s, "EA", "EM", "EK", "IA", "IM", "IK") && (!skipIntern || !StringUtils.equalsAny(s, "IA", "IM", "IK")))
```

Comment:
For clarity, consider using the parameter from above to initialize the matching list:

```java
String[] expectedContactTypes = 
    skipInternalContacts ?
    new String[] { "EA", "EM", "EK" } :
    new String[] { "EA", "EM", "EK", "IA", "IM", "IK" };
```

Reply:
Updated